binderhub/static/js/index.js

```diff
@@ -272,6 +250,10 @@ function indexMain() {
     // setup badge dropdown and default values.
     updateUrls();
 
+    loadConfig(function(res) {
+      config_dict = JSON.parse(res);
```

Comment:
Because `loadConfig` is done asynchronously, when `updateRepoText()` is called first time during page load, `config_dict` has the empty value. I think we have to call  `updateRepoText()` here after config_dict is loaded and delete the other call in `indexMain()`.

Reply:
You're right @bitnik, thanks!
I have one question related to storing the config in a global dict though. Are there any reasons for the config to change while binderhub is running (though POST requests to the _config endpoint)? Will this be useful?

Reply:
@bitnik, I ended up checking in `updateRepoText()` if  `config_dict` is empty because I think otherwise there would still have been cases when `updateRepoText()` might have ended up with an empty config dict.
For example, if in a worst case scenario the request to `_config` would have taken a very long time (though I don't see this very likely) and during this time someone would have triggered the event associated with the provider selection (`provider_prefix_sel`), `updateRepoText()` would get in this case an empty dict. What do you think?

Reply:
> Are there any reasons for the config to change while binderhub is running (though POST requests to the _config endpoint)? Will this be useful?
I think we don't need sth like this now. The config is fetched each time the page is loaded and also I think this config will change rarely.